### PR TITLE
Poezio 0.13.1

### PR DESCRIPTION
**Pull request by Flowdalic**

### CODE REVIEW

**Comment by Flowdalic** [comment]
[please reassign]

**Comment by DarthGandalf** [comment]
@Flowdalic prepend `[please reassign]` to the PR title instead.

File: dev-python/slixmpp/slixmpp-1.5.2.ebuild [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+PYTHON_COMPAT=( python3_7 )
```
Comment (thesamesam) [inline dev-python/slixmpp/slixmpp-1.5.2.ebuild:5]: Can you try tests with python 3.8, 3.9 too?

File: dev-python/slixmpp/slixmpp-1.5.2.ebuild [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
```
Comment (thesamesam) [inline dev-python/slixmpp/slixmpp-1.5.2.ebuild:4]: One [blank](https://wiki.gentoo.org/wiki/Project:Proxy_Maintainers/User_Guide/Style_Guide#One_empty_line_below_EAPI.3D_line) line under EAPI.

File: dev-python/slixmpp/slixmpp-1.5.2.ebuild [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+	dev-python/pyasn1-modules[${PYTHON_USEDEP}]
+	dev-python/pyasn1[${PYTHON_USEDEP}]
+"
+DEPEND="
+	dev-python/setuptools[${PYTHON_USEDEP}]
+	net-dns/libidn
+	test? ( $RDEPEND )
+"
+
+python_test() {
```
Comment (thesamesam) [inline dev-python/slixmpp/slixmpp-1.5.2.ebuild:36]: [```distutils_enable_tests setup.py```](https://dev.gentoo.org/~mgorny/python-guide/distutils.html#enabling-tests)

File: net-im/poezio/poezio-0.13.1.ebuild [hunk 4]
```diff
@@ -0,0 +1,50 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+PYTHON_COMPAT=( python3_7 )
```
Comment (thesamesam) [inline net-im/poezio/poezio-0.13.1.ebuild:5]: See comments for other.

File: dev-python/slixmpp/slixmpp-1.5.2.ebuild [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+RDEPEND="
+	dev-python/aiodns[${PYTHON_USEDEP}]
+	dev-python/aiohttp[${PYTHON_USEDEP}]
+	dev-python/pyasn1-modules[${PYTHON_USEDEP}]
+	dev-python/pyasn1[${PYTHON_USEDEP}]
+"
+DEPEND="
+	dev-python/setuptools[${PYTHON_USEDEP}]
+	net-dns/libidn
+	test? ( $RDEPEND )
```
Comment (thesamesam) [inline dev-python/slixmpp/slixmpp-1.5.2.ebuild:33]: ```${RDEPEND}```, and do we definitely [need](https://dev.gentoo.org/~mgorny/python-guide/distutils.html#different-build-system-variations) setuptools in DEPEND? (may be RDEPEND if in entry_points in setup.py, BDEPEND if just generally needed at buildtime which the eclass distutils-r1 does by default).

File: dev-python/slixmpp/slixmpp-1.5.2.ebuild (hunk 1 above)
Comment (Flowdalic) [reply dev-python/slixmpp/slixmpp-1.5.2.ebuild]: Tested with 3.8, some dependencies of slixmpp and poezio do not declare compatibility with python 3.9 (yet).

Comment (thesamesam) [reply dev-python/slixmpp/slixmpp-1.5.2.ebuild]: Can you file bugs for those dependencies if they don't already exist?

File: dev-python/slixmpp/slixmpp-1.5.2.ebuild [hunk 6]
```diff
@@ -0,0 +1,34 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{7,8} )
```
Comment (thesamesam) [inline dev-python/slixmpp/slixmpp-1.5.2.ebuild:6]: 3.6, 3.9 work?

File: net-im/poezio/poezio-0.13.1.ebuild [hunk 7]
```diff
@@ -0,0 +1,51 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{7,8} )
```
Comment (thesamesam) [inline net-im/poezio/poezio-0.13.1.ebuild:6]: 3.6, 3.9 work?

File: net-im/poezio/poezio-9999.ebuild [hunk 8]
```diff
@@ -2,7 +2,8 @@
 # Distributed under the terms of the GNU General Public License v2
 
 EAPI=7
-PYTHON_COMPAT=( python3_6 )
+
+PYTHON_COMPAT=( python3_{7,8} )
```
Comment (thesamesam) [inline net-im/poezio/poezio-9999.ebuild:6]: 3.6, 3.9 work?

**Review by andrey-utkin — CHANGES_REQUESTED** [review]
Hi @Flowdalic , thanks for this PR.

I would suggest @thesamesam to discount the concerns which originate from previous ebuilds which are cloned to make new ones, except for the functionally critical issues. Polishing stuff every time we look at them is nice, but keeping the diffs minimal is important for reviewing stuff quickly and not being bogged down into research. Ebuild changes reviewing is already problematic because of file cloning, so I need to manually checkout the branch and run `diff` in various combinations. Let's not make it worse. This is why I regularly rant that a version bump commit should not be used as an opportunity for all the sweeping changes one can envision. For each little change, make it in a clean workcopy, prepare a commit and write down your justification. If you do that in a 9999 ebuild every time, you will even manage to preserve a deep change history like you have in a typical code git repo.

@Flowdalic two actions please:
* drop that EGIT_REPO_URI change;
* check that `slixmpp>=1.5.0`.

I will push it then as it looks fine.

File: net-im/poezio/poezio-9999.ebuild [hunk 9]
```diff
@@ -12,7 +13,7 @@ LICENSE="ZLIB"
 SLOT="0"
 
 if [[ "${PV}" == "9999" ]]; then
-	EGIT_REPO_URI="https://lab.louiz.org/${PN}/${PN}.git"
+	EGIT_REPO_URI="https://git.poez.io/${PN}.git"
```
[the remark this answers is not in the document]
Comment (Flowdalic) [reply net-im/poezio/poezio-9999.ebuild:16]: Sorry, that is clearly wrong.

File: net-im/poezio/poezio-9999.ebuild [hunk 10]
```diff
@@ -23,14 +24,13 @@ RDEPEND="
 	dev-python/aiodns[${PYTHON_USEDEP}]
 	dev-python/pyasn1-modules[${PYTHON_USEDEP}]
 	dev-python/pyasn1[${PYTHON_USEDEP}]
-	dev-python/slixmpp[${PYTHON_USEDEP}]
+	>=dev-python/slixmpp-1.5.0[${PYTHON_USEDEP}]
```
[the remark this answers is not in the document]
Comment (Flowdalic) [reply net-im/poezio/poezio-9999.ebuild:27]: > Where from did you take `>=1.5.0`?

I assume that at the time I wrote it, it was still `>=1.5.0`. Was changed with https://lab.louiz.org/poezio/poezio/-/commit/976dd2df869b736b3ef9e6be0d1f365181f504ca in May.

File: net-im/poezio/poezio-9999.ebuild [hunk 11]
```diff
@@ -23,14 +24,13 @@ RDEPEND="
...
 "
 
 distutils_enable_tests pytest
 
 DOC_CONTENTS="
 Install these optional runtime dependencies for additional features.
-* dev-python/potr for OTR encryption support.
```
[the remark this answers is not in the document]
Comment (Flowdalic) [reply net-im/poezio/poezio-9999.ebuild:33]: > Why? Have they dropped OTR functionality?

No, but gentoo has last-rited dev-python/potr in https://github.com/gentoo/gentoo/commit/faf686e9ee2deea8d2561cbd63001c7f8cfb5f33

**Comment by andrey-utkin** [comment]
Pushed d299a9d6a0b4e40b2645186d2f900d546ef55d70 ... 1fca18b635202b54a95350f6d74ae322d0f44c05 .
Thanks all!

**Pull request closed without merging**